### PR TITLE
Changes from repository common refactor

### PR DESCRIPTION
**Pull request by Xierumeng**

### CODE REVIEW

File: modules/diversion_waypoints_from_vertices.py [hunk 1]
```diff
@@ -4,64 +4,60 @@
...
-        waypoint_x: location_ground.LocationGround, waypoiny_y: location_ground.LocationGround
+        waypoint_x: location_global.LocationGlobal,
+        waypoint_y: location_global.LocationGlobal,
     ) -> float:
-        latitude_difference: float = waypoint_x.latitude - waypoiny_y.latitude
-        longitude_difference: float = waypoint_x.longitude - waypoiny_y.longitude
+        latitude_difference = waypoint_x.latitude - waypoint_y.latitude
+        longitude_difference = waypoint_x.longitude - waypoint_y.longitude
         return (
             latitude_difference * latitude_difference + longitude_difference * longitude_difference
```
Comment (Xierumeng) [inline modules/diversion_waypoints_from_vertices.py:32]: Change this.

File: path_2024_task_2.py [hunk 2]
```diff
@@ -78,9 +78,10 @@ def main() -> int:
         print("ERROR: Unable to convert waypoints from dict to list")
         return -1
 
-    location_ground_list = list(map(lambda waypoint: waypoint.location_ground, waypoints_list))
-    result, _ = ground_locations_to_kml.ground_locations_to_kml(
-        location_ground_list,
+    # TODO
```
Comment (Xierumeng) [inline path_2024_task_2.py:81]: Address this.

File: modules/load_waypoint_name_to_coordinates_map.py [hunk 3]
```diff
@@ -38,25 +46,36 @@ def load_waypoint_name_to_coordinates_map(
...
-            # Skp header and empty lines
+            # Skip header and empty lines
             parts = line.split(",")
             if line in "name,latitude,longitude,altitude\n" or len(parts) < 4:
                 continue
 
             name, latitude, longitude, altitude = parts
-            name_to_coordinates_and_altitude_map[name] = waypoint.Waypoint(
-                name, float(latitude), float(longitude), float(altitude)
+            result, point = position_global_relative_altitude.PositionGlobalRelativeAltitude.create(
```
Comment (maxlou05) [inline modules/load_waypoint_name_to_coordinates_map.py:72]: Do we not want the name field anymore here?

File: modules/plot_circular_path.py [hunk 4]
```diff
@@ -4,112 +4,128 @@
...
-                    waypoint.location_ground.longitude,
-                    waypoint.altitude,
-                ]
-            )
+    try:
+        with open(filepath, mode="w", encoding="utf-8", newline="") as file:
+            writer = csv.writer(file)
+            writer.writerow(["Latitude", "Longitude", "Altitude"])
+            for waypoint in waypoints:
+                writer.writerow(
```
Comment (maxlou05) [inline modules/plot_circular_path.py:118]: So is waypoint now implied to be without a name?

File: tests/unit/test_plot_circular_path.py [hunk 5]
```diff
@@ -2,126 +2,276 @@
...
-    the x and y directions.
-    A default epsilon of 1e-7 was chosen so that any error within that range is
-    within an error of 1.1 centimeter.
-    """
-    return (
-        abs(point_1.location_ground.latitude - point_2.location_ground.latitude) < epsilon
-        and abs(point_1.location_ground.longitude - point_2.location_ground.longitude) < epsilon
-        and abs(point_1.altitude - point_2.altitude) < epsilon
-    )
+DEFAULT_TOLERANCE = 1e-6
```
Comment (maxlou05) [inline tests/unit/test_plot_circular_path.py:13]: changed from 1e-7 to 1e-6?

**Review by maxlou05 — COMMENTED** [review]
Reviewed

File: tests/unit/test_plot_circular_path.py (hunk 5 above)
Comment (Xierumeng) [reply tests/unit/test_plot_circular_path.py:13]: Pytest default is 1e-6 .

**Review by maxlou05 — APPROVED** [review]
Approved

**Pull request merged**